### DISCUSSION TITLE
My commits aren't being attributed to my account

### DISCUSSION THREAD

**Discussion by cybertheory** [community/community#24008] [New to GitHub]
<p>Hi, I looked at my recent commits in some of my repos and noticed that Github has not been linking my account with the commits I make. Instead it identifies the author of the commits with my name (not a clickable username).</p>
<p>I am unable to interact with the connection to my account. This has led me to believe that Github is using some sort of default UI element and not actually linking my work to my account.</p>
<p>Does anybody else have this issue? How can I fix this?</p>

**Comment by cybertheory** [comment]
<p>some pictures for context:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/0/10c1b1b81f0daae917f4d1e6fb0604e78d151181.png" data-download-href="/uploads/short-url/2oeCTZMiOq10DLVzQeQOgB19oNr.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/27149047/181080617-98e32b26-4a9c-4ae9-9599-d8e6afa3928d.png" alt="image" data-base62-sha1="2oeCTZMiOq10DLVzQeQOgB19oNr" width="690" height="119" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/0/10c1b1b81f0daae917f4d1e6fb0604e78d151181_2_690x119.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/0/10c1b1b81f0daae917f4d1e6fb0604e78d151181_2_1035x178.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/0/10c1b1b81f0daae917f4d1e6fb0604e78d151181.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/0/10c1b1b81f0daae917f4d1e6fb0604e78d151181_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1157×200 16.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by cybertheory** [comment]
<p>Here it was when it worked:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/1/f10fd8e25937a56d33d810ab579664fa04c718f0.png" data-download-href="/uploads/short-url/yox1pRaKVd5POjKRN28JW5YlD7G.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/27149047/181080618-5b43b5a4-0866-418f-9ca7-7599e0a3360a.png" alt="image" data-base62-sha1="yox1pRaKVd5POjKRN28JW5YlD7G" width="690" height="81" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/1/f10fd8e25937a56d33d810ab579664fa04c718f0_2_690x81.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/1/f10fd8e25937a56d33d810ab579664fa04c718f0_2_1035x121.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/f/1/f10fd8e25937a56d33d810ab579664fa04c718f0.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/1/f10fd8e25937a56d33d810ab579664fa04c718f0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1161×137 9.18 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by cybertheory** [comment]
<p>I fixed my issue by changing the user.name variable on my local git. I am wondering now: how can I go back and claim the commits I made earlier under a different username?</p>

**Comment by mpboom** [comment, marked as answer]
<p>Hey there <a class="mention" href="/cybertheory">@cybertheory</a> and welcome!</p>
<p>Each Git commit has an associated e-mail address. It’s set using <code>git config —global user.email</code> I believe.</p>
<p>GitHub will attribute a commit to you if you have the e-mail address used for the commit in your account. You can add any e-mail address at any time to your account to retroactively correct commit attribution from past commits.</p>
<p><a href="https://docs.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user" rel="noopener nofollow ugc">Read more on that here.</a></p>
